x/evm/statedb/native.go

```diff
@@ -17,4 +18,5 @@ func (native nativeChange) Dirtied() *common.Address {
 
 func (native nativeChange) Revert(s *StateDB) {
 	s.restoreNativeState(native.snapshot)
+	s.nativeEvents = s.nativeEvents[:len(s.nativeEvents)-native.events]
```

Comment:
if not assume out of bound
```suggestion
	if len(s.nativeEvents)-native.events >= 0 {
		s.nativeEvents = s.nativeEvents[:len(s.nativeEvents)-native.events]
	}
```

Reply:
if `native.events > len(s.nativeEvents)`, there's something wrong.

Reply:
if `native.events > len(s.nativeEvents)`, there's something wrong.